### PR TITLE
[package:flutter_lints] update the dep on `package:lints` to 5.0.0 and the sdk dependency to 3.5.0

### PR DESCRIPTION
**Pull request by devoncarew**
- rev `package:flutter_lints` to 5.0.0 in prep for publishing
- update our dep on package:lints to the latest stable published version (`5.0.0`)
- update our sdk dep to reflect the package:lints one
- update the changelog to reflect the new changes brought in from package:lints

Closes https://github.com/dart-lang/lints/issues/205.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

**Comment by devoncarew** [comment]
@stuartmorgan for review (and cc @Piinks and @goderbauer for fyi)

File: packages/flutter_lints/CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
-
-* Removed the following lints:
-    * `prefer_const_constructors` (see https://github.com/dart-lang/lints/issues/205)
+## 5.0.0
+
+* Updates `package:lints` dependency to version 5.0.0, with the following changes:
+    * adds `invalid_runtime_check_with_js_interop_types`
+    * adds `unnecessary_library_name`
+    * removes `avoid_null_checks_in_equality_operators`
+* Removed the following lints (see https://github.com/dart-lang/lints/issues/205):
```
Comment (stuartmorgan) [inline packages/flutter_lints/CHANGELOG.md:7]: Removes

**Comment by stuartmorgan** [comment]
> rev to 5.0.0 in prep for publishing

In flutter/packages this doesn't prepare for publishing, landing this PR actually publishes the package. We generally give the PR a name describing the change itself rather than the version change (e.g., "[flutter_lints] Update `lints` to 5.0 and adjust additional lints".

**Comment by devoncarew** [comment]
> We generally give the PR a name describing the change itself rather than the version change (e.g., "[flutter_lints] Update `lints` to 5.0 and adjust additional lints") to make the history more informative.

👍 ; updated the title to `[package:flutter_lints] update the dep on package:lints to 5.0.0 and the sdk dependency to 3.5.0`.

**Review by stuartmorgan — APPROVED** [review]
LGTM with changelog style nit

File: packages/flutter_lints/CHANGELOG.md (hunk 1 above)
Comment (devoncarew) [inline packages/flutter_lints/CHANGELOG.md:7]:
```suggestion
* Removes the following lints (see https://github.com/dart-lang/lints/issues/205):
```

**Pull request merged**